packages/eslint-config/.eslintrc.js

```diff
@@ -46,14 +45,45 @@ module.exports = {
     "import/no-cycle": ["error", { maxDepth: Infinity }],
     "import/no-useless-path-segments": "error",
     "import/no-relative-parent-imports": "error",
+    "import/no-deprecated": "error",
     "import/newline-after-import": "error",
+    "no-null/no-null": "error",
+    "react/no-danger": "error",
     "prettier/prettier": [
       "error",
       { trailingComma: "all" },
-      { usePrettierrc: false },
+      { usePrettierrc: false }
     ],
+    "@typescript-eslint/consistent-type-assertions": [
+      "error",
+      {
+        assertionStyle: "as"
+      }
+    ],
+    "@typescript-eslint/no-this-alias": "error",
+    "@typescript-eslint/no-extraneous-class": "error",
     "@typescript-eslint/no-use-before-define": ["error", { functions: false }],
     "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/no-unused-vars": "error",
+    curly: ["error", "multi-line", "consistent"],
+    "prefer-const": "error",
+    "dot-notation": "error",
     "sort-imports": ["error", { ignoreDeclarationSort: true }],
-  },
+    "guard-for-in": "error",
+    "max-statements": ["error", 12],
+    "no-bitwise": "error",
+    "no-caller": "error",
+    "no-eval": "error",
+    "no-shadow": "error",
+    "no-sequences": "error",
+    "no-new-func": "error",
+    "no-template-curly-in-string": "error",
+    "no-return-await": "error",
```

Comment:
Worth re-visiting this one. From JetBrains: 
> Missing await for an async function call results in having a Promise where a value was expected.  When the 'Report for promises in return statements' checkbox is checked, this inspection will also suggest adding 'await' in return statements. While this is generally not necessary, it gives two main benefits. The first one is that you won't forget to add 'await' when surrounding your code with try-catch. The second one is that having explicit 'await' helps V8 runtime to provide async stack traces.

The second-one won't trigger this rule, but the first one is applicable. Let's see how it goes for day-to-day debugging/production error logs and revisit if necessary